corehq/apps/users/tasks.py

```diff
@@ -123,33 +123,49 @@ def tag_forms_as_deleted_rebuild_associated_cases(user_id, domain, form_id_list,
         _rebuild_case_with_retries.delay(domain, case_id, detail)
 
 
-@task(queue='background_queue', ignore_result=True, acks_late=True)
-def tag_system_forms_as_deleted(domain, deleted_forms, deleted_cases, deletion_id, deletion_date):
-    form_ids_to_delete = set()
-    for case_id in deleted_cases:
+def _get_forms_to_modify(domain, modified_forms, modified_cases, is_deletion):
+    """Used on user.retire() and user.unretire()
+
+    Returns a list of IDs of forms which only modify the cases passed in and
+    which aren't already listed in `modified_forms`.
+    """
+    form_ids_to_modify = set()
+    for case_id in modified_cases:
         xform_ids = CaseAccessors(domain).get_case_xform_ids(case_id)
```

Comment:
The tests I wrote currently fail on the couch backend because the couch version of this method doesn't return the system forms.  Any ideas on how to get those for couch backends?

Reply:
Strange, not sure why that would be. The list of forms is pulled from a couch view so I'd check the view to see what's going on (`form_case_index/form_case_index`)

Reply:
Looks like that view excludes `XFormInstance-Deleted` documents, whereas the corresponding SQL accessor _does_ include deleted results.  That's an unfortunate discrepancy.  I'm not sure if it's possible to get the deleted forms which updated a given case in a performant fashion.  That would probably require a new couch view (or a modification to this one).  I'm inclined to merge this change as-is, so it'll work for SQL domains, at least, although I could be talked out of that if you think the lack of parity is a blocker.